src/Robo/Commands/Generate/AliasesCommand.php

```diff
@@ -146,19 +148,20 @@ protected function loadCloudApiConfigFile() {
    *
    * @return array
    *   Returns credentials as array on success.
+   * @throws BltException
    */
   protected function askForCloudApiCredentials() {
     $this->say("You may generate new API tokens at <comment>https://cloud.acquia.com/app/profile/tokens</comment>");
     $key = $this->askRequired('Please enter your Acquia cloud API key:');
     $secret = $this->askRequired('Please enter your Acquia cloud API secret:');
-    do {
-      $this->setCloudApiClient($key, $secret);
-      $cloud_api_client = $this->getCloudApiClient();
-    } while (!$cloud_api_client);
-    $config = array(
+
+    // Attempt to set client to check credentials (throws exception on failure).
```

Comment:
Hmm, this removes the loop entirely, which perhaps is probably okay, but I am curious as to why we had to add a loop in the first place?

Reply:
Grasmick originally wrote this command along with the loop here :https://github.com/acquia/blt/pull/2168

There's no documentation there as to why a loop was used. I vaguely recall something about authentication sometimes failing on the first try due to problems with the hmac library, but I think all of those issues have since been resolved. I tested this extensively and didn't experience any problems authenticating on the first try (indeed, if the API was behaving in a stateful way like that it would be a much bigger issue).

Reply:
Looks good to me then!